homeassistant/components/husqvarna_automower/__init__.py

```diff
@@ -0,0 +1,52 @@
+"""The Husqvarna Automower integration."""
+
+import contextlib
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.config_entry_oauth2_flow import (
+    async_get_config_entry_implementation,
+)
+
+from .const import DOMAIN, PLATFORMS
+from .coordinator import AutomowerDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up this integration using UI."""
+    hass.data.setdefault(DOMAIN, {})
+    implementation = await async_get_config_entry_implementation(hass, entry)
+    entry_dict = entry.as_dict()
```

Comment:
We can get the token from the config entry data. We don't need to use `entry_as_dict`.